components/infobox/wikis/arenaofvalor/infobox_player_custom.lua

```diff
@@ -0,0 +1,137 @@
+local Cell = Widgets.Cell
+local Title = Widgets.Title
+local Center = Widgets.Center
+
+local _pagename = mw.title.getCurrentTitle().prefixedText
+local _role
+local _role2
+local _EMPTY_AUTO_HISTORY = '<table style="width:100%;text-align:left"></table>'
+local _SIZE_HERO = '25x25px'
+
+local CustomPlayer = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _player
+
+function CustomPlayer.run(frame)
+	local player = Player(frame)
+	_player = player
+	_args = player.args
+
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'history' then
+		local manualHistory = _args.history
+		local automatedHistory = TeamHistoryAuto._results({
+			convertrole = 'true',
+			player = _pagename
+		}) or ''
+		automatedHistory = tostring(automatedHistory)
+		if automatedHistory == _EMPTY_AUTO_HISTORY then
+			automatedHistory = nil
+		end
+
+		if not (String.isEmpty(manualHistory) and String.isEmpty(automatedHistory)) then
+			return {
+				Title{name = 'History'},
+				Center{content = {manualHistory}},
+				Center{content = {automatedHistory}},
+			}
+		end
+	elseif id == 'region' then return {}
+	elseif id == 'role' then
+		_role = Role.run({role = _args.role})
+		_role2 = Role.run({role = _args.role2})
+		return {
+			Cell{name = 'Role(s)', content = {_role.display, _role2.display}}
+		}
+	end
+	return widgets
+end
+
+
+function CustomInjector:addCustomCells(widgets)
+	-- Signature Heroes
+	local heroIcons = Array.map(Player:getAllArgsForBase(_args, 'hero'),
+		function(hero, _)
+			local standardizedHero = HeroNames[hero:lower()]
+			if not standardizedHero then
+				-- we have an invalid hero entry
+				-- add warning (including tracking category)
+				table.insert(
+					_player.warnings,
+					'Invalid hero input "' .. hero .. '"[[Category:Pages with invalid hero input]]'
+				)
+			end
+			return HeroIcon.getImage{standardizedHero or hero, size = _SIZE_HERO}
+		end
+	)
+
+	if Table.isNotEmpty(heroIcons) then
+		table.insert(
+			widgets,
+			Cell{
+				name = #heroIcons > 1 and 'Signature Heroes' or 'Signature Hero',
+				content = {
+					table.concat(heroIcons, '&nbsp;')
+				}
+			}
+		)
+	end
+
+	return widgets
+end
+
+function CustomPlayer:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomPlayer:adjustLPDB(lpdbData)
+	lpdbData.extradata.isplayer = _role.isPlayer or 'true'
+	lpdbData.extradata.role = _role.role
+	lpdbData.extradata.role2 = _role2.role
+
+	-- store signature heroes with standardized name
+	for heroIndex, hero in ipairs(Player:getAllArgsForBase(_args, 'hero')) do
+		lpdbData.extradata['signatureHero' .. heroIndex] = HeroNames[hero:lower()]
+	end
+
+	local region = Region.run({region = _args.region, country = _args.country})
+	if type(region) == 'table' then
+		lpdbData.region = region.region
+	end
```

Comment:
you store region data here but kicked the display above

redundant? commons should store the same

Reply:
Ok, I've added a revision that removes the region calls 

anyway does this line need to be add ? 

**lpdbData.region = Template.safeExpand(mw.getCurrentFrame(), 'Player region', {_args.country})**

(this is to add at line 126 in the newest revision, if it needs to)